CustomerService/Service/CustomerDataService.cs

```diff
@@ -23,23 +24,65 @@ public CustomerDataService(ICustomerRepository _customerRepository, IDomainEvent
         }
         public Customer CreateCustomer(String name, Money creditLimit)
         {
-            Customer customer;
+            Customer customer = new Customer();
             using (var scope = new TransactionScope())
             {
-                ResultsWithEvents customerWithEvents = Create(name, creditLimit);
-                customer = customerRepository.InsertCustomer(customerWithEvents.Customer);
+                ResultsWithEvents customerWithEvents = customer.Create(name, creditLimit);
+                customer = customerRepository.Add(customerWithEvents.Customer);
                 domainEventPublisher.Publish(typeof(Customer).Name, customer.Id, customerWithEvents.Events);
                 scope.Complete();
                 return customer;
             }
         }
-        public static ResultsWithEvents Create(String name, Money creditLimit)
+        public void ReserveCredit(long orderId, long customerId, Money orderTotal)
         {
-            Customer customer = new Customer(name, creditLimit);
-            var customerCreatedEvent = new CustomerCreatedEvent(customer.Name, customer.CreditLimit);
-            List<IDomainEvent> eventList = new List<IDomainEvent>();
-            eventList.Add(customerCreatedEvent);
-            return new ResultsWithEvents(customer, eventList);
+            Customer customer = customerRepository.FindById(customerId);
+            using (var scope = new TransactionScope())
```

Comment:
Methods that are invoked when handling a message should not manage transactions.
The Tram framework should have started a transaction already

Reply:
Done